gourmet/gtk/recipe.py

```diff
@@ -0,0 +1,787 @@
+        model, selected = tree.get_selection().get_selected()
+        if selected:
+            row = model.get_value(selected, 4)
+            if isinstance(row, _IngredientGroup):
+                group_model = groups.get_model()
+                group_model.remove(row.iter)
+                groups.set_property('sensitive', len(group_model) > 1)
+                if not groups.get_active_iter():
+                    groups.set_property('active', 0)
+            elif row:
+                deleted.append(row)
+            model.remove(selected)
+            return True
+    return False
+
+
+# TODO: Make non-name columns non-activatable for ingredient groups
+# This probably requires implementing CellRenderer subclasses that don't return
+# a CellEditable if the row is an ingredient group. Cells are currently still
+# activatable, meaning the user can enter edits, but any edits are rejected.
+
+def _edit_ingredient(renderer, path, text, model, column, db=None, groups=None):
+    # Reject edits if the edited row is an ingredient group name unless it's to
+    # change the name
+    iter = model.get_iter_from_string(path)
+    obj = model.get_value(iter, 4)
+    is_group = isinstance(obj, _IngredientGroup)
+    if 2 == column or not is_group:
+        model.set_value(iter, column, text)
+        if 2 == column:
+            if is_group:
+                assert groups
+                group_iter = obj.iter
+                groups.get_model().set_value(group_iter, 0, text)
+            else:
+                assert db
+                recipe_id = _check_if_recipe(db, text)
+                model.set_value(iter, 5, recipe_id)
+
+
+def _toggle_ingredient(renderer, path, model, column):
+    # Reject edits if the edited row is an ingredient group name
+    iter = model.get_iter_from_string(path)
+    if not isinstance(model.get_value(iter, 4), _IngredientGroup):
+        value = model.get_value(iter, column)
+        model.set_value(iter, column, not value)
+
+
+def _add_group(parent, ingredients):
+    dialog = _Gtk.Dialog(
+        title = _('New Ingredient Group'),
+        transient_for = parent,
+        destroy_with_parent = True
+    )
+
+    main = _Gtk.Box(
+        orientation = _Gtk.Orientation.HORIZONTAL,
+        margin_bottom = 6,
+        spacing = 6
+    )
+    main.pack_start(_Gtk.Label(label=_('Group Name:')), False, False, 0)
+
+    entry = _Gtk.Entry()
+    def enter_group(_):
+        if entry.get_text():
+            dialog.response(_Gtk.ResponseType.OK)
+    entry.connect('activate', enter_group)
+    main.pack_start(entry, True, True, 0)
+
+    content = dialog.get_content_area()
+    content.set_property('margin', 6)
+    content.pack_start(main, True, True, 0)
+
+    dialog.add_button('Cancel', _Gtk.ResponseType.CANCEL)
+    dialog.add_button('Ok', _Gtk.ResponseType.OK)
+
+    dialog.show_all()
+    response = dialog.run()
+
+    if _Gtk.ResponseType.OK == response:
```

Comment:
That's a trick that I do a lot in C/C++, but PEP8 enforces 
```suggestion
    if response ==_Gtk.ResponseType.OK:
```
It also applies to other occurrences throughout this file.

Reply:
I'm not getting any errors about this from pycodestyle (formerly pep8). The rest is whitespace and line length issues which I can certainly fix.

Reply:
I use `flake8`. I stand corrected though: it seem to be valid!   
I would still ask you to stick to the more conventional `val == const`